src/tools/check_brokenURL.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/python
+
+import re
+import os
+import sys
+import getopt
+import requests
+import argparse
+
+def get_filepaths(directory):
+    file_paths = []  
+    for root, directories, files in os.walk(directory):
+        for filename in files:
+            filepath = os.path.join(root, filename)
+            file_paths.append(filepath)
+    return file_paths
+
+def checkBrokenLink(url):
+    if requests.get(url).ok == True:
+        return False
+    else:
+        return True
+
+def scrapLink(filePath):
+    fo=open(filePath,'r')
+    data=fo.read()
+    urls = re.findall('http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\(\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+', data)
+    for url in urls:
+        if checkBrokenLink(url):
+            print("Found broken link : "+url+"\n\t in file: "+fo.name+"\n\t at:"+os.path.dirname(filePath)+"\n \n")
```

Comment:
we check broken links with only 1 thread, and we try to `get` every link(not `head` at first). it would be better if you could test it with whole ceph source tree and give us some information about time usage.